zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -165,14 +165,16 @@ impl UpdateWith<PreparedBlock> for Chain {
             .zip(transaction_hashes.iter().cloned())
             .enumerate()
         {
+            use transaction::Transaction::*;
             let (inputs, shielded_data, joinsplit_data) = match transaction.deref() {
-                transaction::Transaction::V4 {
+                V4 {
                     inputs,
                     shielded_data,
                     joinsplit_data,
                     ..
                 } => (inputs, shielded_data, joinsplit_data),
-                _ => unreachable!(
+                V5 { .. } => unimplemented!("v5 transaction format as specified in ZIP-225"),
+                V1 | V2 | V3 => unreachable!(
```

Comment:
We need to handle v5 transactions in the non-finalized state.

I also removed the wildcard match, because they make it harder to do upgrades like this.